homeassistant/components/sensor/wunderground.py

```diff
@@ -666,8 +679,12 @@ def __init__(self, hass: HomeAssistantType, rest, condition):
         self._entity_picture = None
         self._unit_of_measurement = self._cfg_expand("unit_of_measurement")
         self.rest.request_feature(SENSOR_TYPES[condition].feature)
+        current_ids = set(hass.states.async_entity_ids(sensor.DOMAIN))
```

Comment:
Why would you generate this in every constructor call? Just do it once and pass it into the constructor

Reply:
Hmm, true... that's inefficient indeed.

(on a side note: all entities (from all entity platforms) do not cache this)